blocks/api/raw-handling/special-comment-converter.js

```diff
@@ -19,10 +19,27 @@ const { COMMENT_NODE } = window.Node;
  * @return {void}
  */
 export default function( node ) {
-	if (
-		node.nodeType !== COMMENT_NODE ||
-		node.nodeValue.indexOf( 'more' ) !== 0
-	) {
+	if ( node.nodeType !== COMMENT_NODE ) {
+		return;
+	}
+
+	if ( node.nodeValue.indexOf( 'nextpage' ) === 0 ) {
+		// Conjure up a custom Nextpage element
+		const nextPage = createNextpage();
+
+		// Append it to the top level for later conversion to blocks
+		let parent = node.parentNode;
+		while ( parent.nodeName !== 'BODY' ) {
+			parent = parent.parentNode;
+		}
+
+		parent.appendChild( nextPage );
+		node.parentNode.removeChild( node );
+
+		return;
```

Comment:
Instead of repeating the parent finding and the node operations for both comment types, we could refactor this. The simplest way I can think of may be:

```js
let newNode;

if ( foundNextpage ) {
  newNode = createNextpage();
} else if ( foundMore ) {
  newNode = createMore( ...args );
}

if ( newNode ) {
  let parent = …; while ( … ) parent = parent.Node;
  parent.appendChild( … ); node.parentNode.removeChild( … );
  return;
}
```

or, actually better:

```js
if ( foundNextpage ) {
  appendToBody( node, createNextpage() );
  remove( node );
  return;
}
if ( foundMore ) {
  appendToBody( node, createMore( … ) );
  remove( node );
}

function appendToBody( processedNode, newNode ) {
  // Append it to the top level for later conversion to blocks
  let parent = node.parentNode;
  while ( parent.nodeName !== 'BODY' ) {
    parent = parent.parentNode;
  }

  parent.appendChild( nextPage );
}

function remove( node ) {
  node.parentNode.removeChild( node );
}
```